### PR TITLE
reduces the response time when close preferences

### PR DESCRIPTION
**Pull request by tanojaja**

### CODE REVIEW

File: src/js/controllers/topbar.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-angular.module('copayApp.controllers').controller('topbarController', function(go) { 
+angular.module('copayApp.controllers').controller('topbarController', function(go, $rootScope) {
 
   this.goHome = function() {
     go.walletHome();
   };
 
   this.goPreferences = function() {
-    go.preferences();
+    $rootScope.$emit('Local/GoPreferences');
```
Comment (matiu) [inline src/js/controllers/topbar.js]: events should be events and not functions.

File: src/js/controllers/index.js [hunk 2]
```diff
@@ -935,8 +936,10 @@ angular.module('copayApp.controllers').controller('indexController', function($r
       });
     });
   }
+
   self.showAllHistory = function() {
     self.historyShowShowAll = false;
+    self.shouldHideShowAll = true;
```
Comment (matiu) [inline src/js/controllers/index.js]: why there are two variables:

 - historyShowShowAll
and
 - shouldHideShowAll 

isn't that the same?

File: src/js/controllers/index.js [hunk 3]
```diff
@@ -926,7 +926,7 @@ angular.module('copayApp.controllers').controller('indexController', function($r
         if (walletId == profileService.focusedClient.credentials.walletId) {
           self.completeHistory = newHistory;
           self.txHistory = newHistory.slice(0, self.historyShowLimit);
-          self.historyShowShowAll = newHistory.length >= self.historyShowLimit;
+          self.historyShowShowAll = newHistory.length > self.historyShowLimit;
```
Comment (tanojaja) [inline src/js/controllers/index.js]: "Show all" appears only if strictly greater than the limit

**Comment by tanojaja** [comment]
Updated

**Comment by tanojaja** [comment]
Updated. Thanks for comment!

**Comment by cmgustavo** [comment]
ACK

**Pull request merged**